ApplicationLibCode/GeoMech/OdbReader/RifInpReader.cpp

```diff
@@ -318,8 +319,46 @@ std::vector<size_t> RifInpReader::readElementSet( std::istream& stream )
         auto parts = RiaStdStringTools::splitString( line, ',' );
         for ( auto part : parts )
         {
-            int elementId = RiaStdStringTools::toInt( part ) - 1;
-            elementSet.push_back( elementId );
+            std::string trimmedPart = RiaStdStringTools::trimString( part );
+
+            if ( !trimmedPart.empty() )
+            {
+                int elementId = RiaStdStringTools::toInt( trimmedPart ) - 1;
+                elementSet.push_back( elementId );
+            }
+        }
+    }
+
+    return elementSet;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+std::vector<size_t> RifInpReader::readElementSetGenerate( std::istream& stream )
+{
+    std::vector<size_t> elementSet;
+
+    // Read until we find a new section (which should start with a '*').
+    while ( stream.peek() != '*' && stream.peek() != EOF )
+    {
+        // Read entire line of comma-separated values
+        std::string line;
+        std::getline( stream, line );
+
+        // Process the comma-separated values
+        auto parts = RiaStdStringTools::splitString( line, ',' );
+        if ( parts.size() >= 3 )
+        {
+            int firstElement = RiaStdStringTools::toInt( parts[0] );
+            int lastElement  = RiaStdStringTools::toInt( parts[1] );
+            int increment    = RiaStdStringTools::toInt( parts[2] );
```

Comment:
Check if increment > 0, to avoid infinite loop?

Reply:
Fixed!